cl/corpus_importer/management/commands/financial_disclosures.py

```diff
@@ -0,0 +1,344 @@
+import json
+from typing import Dict
+from urllib.parse import quote
+
+import requests
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django.db import transaction
+
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.lib.crypto import sha1
+from cl.people_db.models import (
+    FinancialDisclosure,
+    Person,
+    Investment,
+    Agreements,
+    Debt,
+    Gift,
+    Reimbursement,
+    NonInvestmentIncome,
+    SpouseIncome,
+    Positions,
+)
+from cl.scrapers.transformer_extractor_utils import get_page_count
+
+
+def check_if_in_system(sha1_hash: str) -> bool:
+    """Check if pdf bytes hash sha1 in cl db.
+
+    :param sha1_hash: Sha1 hash
+    :return: Whether PDF is in db.
+    """
+    disclosures = FinancialDisclosure.objects.filter(pdf_hash=sha1_hash)
+    if len(disclosures) > 0:
+        logger.info("PDF already in system")
+        return True
+    return False
+
+
+def extract_content(pdf_bytes: bytes) -> Dict:
+    """Extract the content of the PDF.
+
+    Attempt extraction using multiple methods if necessary.
+
+    :param pdf_bytes: The byte array of the PDF
+    :return:The extracted content
+    """
+    logger.info("Beginning Extraction")
+
+    # Extraction takes between 7 seconds and 80 minutes for super
+    # long Trump extraction with ~5k investments
+    extractor_response = requests.post(
+        settings.BTE_URLS["extract-disclosure"],
+        files={"pdf_document": ("file", pdf_bytes)},
+        timeout=60 * 120,
+    )
+
+    if (
+        extractor_response.status_code != 200
+        or extractor_response.json()["success"] is False
+    ):
+        # Try second method
+        logger.info("Attempting second extraction")
+        extractor_response = requests.post(
+            settings.BTE_URLS["extract-disclosure-jw"],
```

Comment:
If we know this is the judicial watch PDF, why not just call this in the first place?

Reply:
We dont just call judicial watch, because the other method _can_ still work on better quality judicial watch pdfs.  This method is faster and probably more accurate so its worth atleast testing that method first and using the second as a backup.